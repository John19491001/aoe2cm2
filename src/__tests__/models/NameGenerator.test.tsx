import {expect, it} from 'vitest';
import NameGenerator from "../../util/NameGenerator";

it('NameGenerator throws error when initialized with 0', () => {
    expect(() => {
        new NameGenerator(0);
    }).toThrowError('LCG must not be initialized with 0');
});

it('NameGenerator yields deterministic results', () => {
    const nameGenerator = new NameGenerator(14);
    const values = [];
    for (let i = 0; i < 10; i++) {
        values.push(nameGenerator.nextName());
    }
    expect(values).toMatchSnapshot();
});
