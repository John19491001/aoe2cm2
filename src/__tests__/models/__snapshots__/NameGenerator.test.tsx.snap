// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NameGenerator yields deterministic results 1`] = `
[
  "King Alfonso",
  "Yaroslav the Wise",
  "Lothair II",
  "King Alfonso",
  "Henry Bolingbroke",
  "Nasiruddin Mahmud Shah",
  "Waterlily Jaguar",
  "Maxixca",
  "Avitus",
  "Frumentius",
]
`;
